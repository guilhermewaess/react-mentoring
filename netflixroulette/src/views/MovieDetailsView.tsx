import * as React from 'react';
import {
  BrowserRouter as Router,
  Route,
  RouteComponentProps,
  RouteProps,
} from 'react-router-dom';

import MovieDetails from '../containers/MovieDetails';

export default ({ match }: RouteComponentProps<RouteProps>) => (
  <Router>
    <Route exatc={true} path={`${match.url}/:id`} component={MovieDetails} />
  </Router>
);
