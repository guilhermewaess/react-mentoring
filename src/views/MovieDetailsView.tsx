import * as React from 'react';
import { Route, RouteComponentProps, RouteProps } from 'react-router-dom';
import MovieDetails from '../containers/MovieDetails';

export default ({ match }: RouteComponentProps<RouteProps>) => (
  <Route exatc={true} path={`${match.url}/:id`} component={MovieDetails} />
);
