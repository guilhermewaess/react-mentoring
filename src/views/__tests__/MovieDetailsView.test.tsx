import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import MovieDetailsView from './../MovieDetailsView';


describe('MovieDetailsView', () => {
  let component: ShallowWrapper;
  let routeProps: any;

  const helper = {
    createRouteProps() {
      return {
        match: { url: '/baseUrl' }
      }
    }
  }

  beforeEach(() => {
    routeProps = helper.createRouteProps();
    component = shallow(<MovieDetailsView {...routeProps} />)
  });
  describe('when construct', () => {
    it('should render', () => {
      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });
});