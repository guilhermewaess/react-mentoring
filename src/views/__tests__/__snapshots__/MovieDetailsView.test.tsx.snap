// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsView when construct should render 1`] = `
<Route
  component={[Function]}
  exatc={true}
  path="/baseUrl/:id"
/>
`;
